File: python/l00.exit.py
#!/usr/bin/python
import sys

# illustrative variables
ARG_COUNT   = len(sys.argv) - 1 # get num of real arguments
SCRIPT_NAME = sys.argv[0]       # get name of script
EX_USAGE    = 64;               # status for command line usage error
EX_OK       = 0;                # status for successful termination

# method (subroutine) to output usage message to stderr
def usage_message ():
   sys.stderr.write("\nYou need to enter two or more numbers: \n\n")
   sys.stderr.write("   Usage: %s [num1] [num2] [num3]...\n\n" % SCRIPT_NAME)
   sys.exit(EX_USAGE)

# method (subroutine) to sum numbers from array
def addNums (numbers):
   sum = 0                      # initialize starting sum value
   for num in numbers:          # collection loop toprocess numbers
     sum += int(num)            # sum up nums
   print "The summation is: %d" % sum # output results
   sys.exit(EX_OK)

# check for only 2 arguments
if ARG_COUNT < 1:
   # ouptut usage statement to standard error
   usage_message()
else:
   # call addNums to process arguments
   addNums(sys.argv[1:])

File: ruby/l00.exit.rb
#!/usr/bin/ruby

arg_count    = ARGV.length # get num of args
@script_name = $0          # get script name

# addNums method (subroutine)
def addNums (numbers)
   sum = 0                               # initialize starting sum value
   numbers.each {|num| sum += num.to_i } # sum up nums
   puts "The summation is: #{sum}"       # output results
end

# addNums method (subroutine)
def usage_message ()
  # output usage statement to standard error
  $stderr.puts "\nYou need to enter one or more numbers: \n\n"
  $stderr.puts "   Usage: #{@script_name} [num1] [num2] [num3]...\n\n"
end

# check for only 2 arguments
if arg_count < 1 then
   usage_message
else
   addNums(ARGV)
end

